internal/controller/plan/service_descriptor_utils.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/apecloud/kubeblocks/internal/controller/component"
+	intctrlutil "github.com/apecloud/kubeblocks/internal/controllerutil"
+)
+
+func GenServiceReferences(reqCtx intctrlutil.RequestCtx,
+	cli client.Client,
+	cluster *appsv1alpha1.Cluster,
+	clusterCompDef *appsv1alpha1.ClusterComponentDefinition,
+	clusterCompSpec *appsv1alpha1.ClusterComponentSpec,
+) (map[string]*appsv1alpha1.ServiceDescriptor, error) {
+	if cluster == nil || clusterCompDef == nil || clusterCompSpec == nil {
+		return nil, nil
+	}
+
+	if len(clusterCompDef.ServiceRefDeclarations) == 0 {
+		return nil, nil
+	}
+
+	serviceReferences := make(map[string]*appsv1alpha1.ServiceDescriptor, len(clusterCompDef.ServiceRefDeclarations))
+	for _, serviceRefDecl := range clusterCompDef.ServiceRefDeclarations {
+		for _, serviceRef := range clusterCompSpec.ServiceRefs {
+			if serviceRef.Name != serviceRefDecl.Name {
+				continue
+			}
+			targetNamespace := cluster.Namespace
+			if serviceRef.Namespace != "" {
+				targetNamespace = serviceRef.Namespace
+			}
+			// if service reference is another KubeBlocks Cluster, then it is necessary to generate a service connection credential from the cluster connection credential secret
+			if serviceRef.Cluster != "" {
+				if serviceRef.Cluster == cluster.Name {
+					return nil, fmt.Errorf("cluster %s cannot reference itself", cluster.Name)
+				}
+				referencedCluster := &appsv1alpha1.Cluster{}
+				if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: targetNamespace, Name: serviceRef.Cluster}, referencedCluster); err != nil {
+					return nil, err
+				}
+				referencedClusterDef := &appsv1alpha1.ClusterDefinition{}
+				if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Name: referencedCluster.Spec.ClusterDefRef}, referencedClusterDef); err != nil {
+					return nil, err
+				}
+
+				// get the connection credential secret of the referenced cluster
+				var sdName string
+				secretRef := &corev1.Secret{}
+				if serviceRef.ConnectionCredential != "" {
+					sdName = component.GenerateCustomServiceDescriptorName(serviceRef.ConnectionCredential)
+					if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: targetNamespace, Name: serviceRef.ConnectionCredential}, secretRef); err != nil {
+						return nil, err
+					}
+				} else {
+					sdName = component.GenerateDefaultServiceDescriptorName(cluster.Name)
+					secretRefName := component.GenerateConnCredential(referencedCluster.Name)
+					if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: targetNamespace, Name: secretRefName}, secretRef); err != nil {
+						return nil, err
+					}
+				}
+
+				// TODO: Second-stage optimization: Cluster-type references no longer perform conversion on the connection credential field. Instead, the configMap or secret is directly passed through to the serviceDescriptor.
+				sdBuilder := builder.NewServiceDescriptorBuilder(targetNamespace, sdName)
+				sdBuilder.SetServiceKind("")
```

Comment:
Add description in the API that kind and version are not checked for cluster references.

Reply:
> Add description in the API that kind and version are not checked for cluster references.

done